### DISCUSSION TITLE
Painting Textured Rects/Images?

### DISCUSSION THREAD

**Discussion by justinturpin** [emilk/egui#1490] [Q&A] [answered]
Hello! I am trying to make a simple 2d tile-based level editor. For the tile-editing screen, in something like wxWidgets I would have made a Canvas element and then use DrawImage to draw the tiles, along with DrawText and such. I see there is a ui.painter() method for drawing arbitrary shapes but I can't see where I would draw an image? Is this even the right approach for something like this? Thanks for the help!

**Comment by emilk** [comment, marked as answer]
``` rust
                let mut mesh = Mesh::with_texture(texture_id);
                mesh.add_rect_with_uv(rect, uv, tint);
                ui.painter().add(Shape::mesh(mesh));
```

It's a bit of a mouthful – feel free to add a helper method to `Painter` and/or `Mesh` in a PR

**Reply by justinturpin** [reply]
Cool thank you! This will work perfectly
